utilities/network.py

```diff
@@ -1038,23 +1038,146 @@ def get_cluster_cni_type(admin_client):
     return Network(client=admin_client, name="cluster").instance.status.networkType
 
 
-def wait_for_ready_sriov_nodes(snns):
-    for status in (INPROGRESS, SriovNetworkNodePolicy.Status.SUCCEEDED):
+def _node_matches_policy_selector(node, policy):
+    """
+    Check if a node matches the policy's nodeSelector.
+
+    Args:
+        node: Node object
+        policy: SriovNetworkNodePolicy object
+
+    Returns:
+        bool: True if node matches policy selector or if no selector is specified
+    """
+    policy_spec = policy.instance.spec
+    node_selector = getattr(policy_spec, "nodeSelector", {})
+
+    if not node_selector:
+        return True
+
+    node_labels = node.instance.metadata.labels
+    for key, value in node_selector.items():
+        if node_labels.get(key) != value:
+            return False
+
+    return True
+
+
+def _verify_sriov_interface_config(sriov_node_network_state, policy):
+    """
+    Verify that SR-IOV interfaces match the policy configuration.
```

Comment:
Sorry, I do not understand.
There is a controller that takes care of updating the NNCP status per the results on applying it on the nodes.
Are you attempting to duplicate the production logic? Or do I miss something?

Reply:
> It's not an NNCP, it's an SR-IOV policy (SriovNetworkNodePolicy). It doesn't have a `.status` section to reflect the status of its own configuration, so you must verify it yourself via the SNNS.

Thank you for correcting me. I indeed thought about NNCP and not SNNP. I am also surprised it does not have the same status as the NNCP one.

You can resolve this, sorry for the noise.